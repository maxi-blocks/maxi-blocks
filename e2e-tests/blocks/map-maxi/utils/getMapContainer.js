const getMapContainer = async page =>
	page.$('.maxi-map-block .maxi-map-block__container .leaflet-container');

export default getMapContainer;
