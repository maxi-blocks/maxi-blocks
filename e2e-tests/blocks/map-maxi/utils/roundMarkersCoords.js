import { round } from 'lodash';

const roundMarkersCoords = markers => {
	return markers.map(marker => {
		const { latitude, longitude } = marker;

		return {
			...marker,
			latitude: round(latitude),
			longitude: round(longitude),
		};
	});
};

export default roundMarkersCoords;
