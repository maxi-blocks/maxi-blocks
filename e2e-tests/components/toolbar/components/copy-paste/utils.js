// eslint-disable-next-line import/prefer-default-export
export const block = {
	clientId: '4547b7b2-4f1b-4e54-8e37-008d33b29885',
	name: 'maxi-blocks/text-maxi',
	isValid: true,
	attributes: {
		'link-active-palette-color-general': 6,
		'link-active-palette-status-general': true,
		'link-hover-palette-color-general': 6,
		'link-hover-palette-status-general': true,
		'link-palette-color-general': 4,
		'link-palette-status-general': true,
		'link-visited-palette-color-general': 6,
		'link-visited-palette-status-general': true,
		parentBlockStyle: 'light',
		blockStyle: 'maxi-light',
		defaultBlockStyle: 'maxi-def-light',
		extraClassName: '',
		isFirstOnHierarchy: true,
		uniqueID: 'text-maxi-5se8ef1z-u81582',
		customLabel: 'Text',
		textLevel: 'p',
		isList: false,
		typeOfList: 'ul',
		listReversed: 0,
		fullWidth: 'full',
		content:
			'<span class="maxi-text-block--has-custom-format maxi-text-block__custom-format--2">Testing</span> <span class="maxi-text-block--has-custom-format maxi-text-block__custom-format--1">Copy</span> <span class="maxi-text-block--has-custom-format maxi-text-block__custom-format--0">Paste</span>',
		'background-status-hover': false,
		'border-color-general': '',
		'border-style-general': 'dashed',
		'border-status-hover': false,
		'border-top-left-radius-general': 50,
		'border-top-right-radius-general': 50,
		'border-bottom-right-radius-general': 50,
		'border-bottom-left-radius-general': 50,
		'border-sync-radius-general': true,
		'border-unit-radius-general': 'px',
		'border-unit-radius-general-hover': 'px',
		'border-top-width-general': 2,
		'border-right-width-general': 2,
		'border-bottom-width-general': 2,
		'border-left-width-general': 2,
		'border-sync-width-general': true,
		'border-unit-width-general': 'px',
		'border-palette-color-general': 2,
		'border-palette-color-general-hover': 6,
		'border-palette-status-general': true,
		'border-palette-status-general-hover': true,
		'box-shadow-color-general': 'var(--maxi-shadow-color)',
		'box-shadow-horizontal-general': 5,
		'box-shadow-vertical-general': 5,
		'box-shadow-blur-general': 0,
		'box-shadow-spread-general': 0,
		'box-shadow-status-hover': false,
		'box-shadow-color-general-hover': '',
		'box-shadow-palette-color-general': 8,
		'box-shadow-palette-color-general-hover': 6,
		'box-shadow-palette-status-general': true,
		'box-shadow-palette-status-general-hover': true,
		'container-size-advanced-options': false,
		'container-min-width-unit-general': 'px',
		'container-max-height-unit-general': 'px',
		'container-height-unit-general': 'px',
		'container-min-height-unit-general': 'px',
		'container-max-width-general': 1170,
		'container-max-width-xxl': 1790,
		'container-max-width-xl': 1170,
		'container-max-width-l': 90,
		'container-max-width-m': 90,
		'container-max-width-s': 90,
		'container-max-width-xs': 90,
		'container-max-width-unit-general': 'px',
		'container-max-width-unit-xxl': 'px',
		'container-max-width-unit-xl': 'px',
		'container-max-width-unit-l': '%',
		'container-max-width-unit-m': '%',
		'container-max-width-unit-s': '%',
		'container-max-width-unit-xs': '%',
		'container-width-l': 1170,
		'container-width-unit-l': 'px',
		'container-width-m': 1000,
		'container-width-unit-m': 'px',
		'container-width-s': 700,
		'container-width-unit-s': 'px',
		'container-width-xs': 460,
		'container-width-unit-xs': 'px',
		'margin-top-general': '10',
		'margin-left-general': '50',
		'margin-sync-general': false,
		'margin-unit-general': 'px',
		'motion-status': false,
		'motion-active-time-line-time': 0,
		'motion-active-time-line-index': 0,
		'motion-transform-origin-x': 'center',
		'motion-transform-origin-y': 'center',
		'motion-preset-status': false,
		'motion-preview-status': false,
		'motion-tablet-status': true,
		'motion-mobile-status': true,
		'padding-top-general': 10,
		'padding-right-general': 10,
		'padding-bottom-general': 10,
		'padding-left-general': 10,
		'padding-sync-general': true,
		'padding-unit-general': 'px',
		'position-general': 'fixed',
		'position-top-general': 50,
		'position-right-general': 50,
		'position-sync-general': false,
		'position-unit-general': '%',
		'size-advanced-options': false,
		'max-width-unit-general': 'px',
		'width-unit-general': 'px',
		'min-width-unit-general': 'px',
		'max-height-unit-general': 'px',
		'height-unit-general': 'px',
		'height-general': 60,
		'min-height-unit-general': 'px',
		'text-alignment-general': 'center',
		'transform-scale-x-general': 112.5,
		'transform-scale-y-general': 107.5,
		'transform-translate-x-unit-general': '%',
		'transform-translate-x-general': -132,
		'transform-translate-y-unit-general': '%',
		'transform-translate-y-general': 52,
		'transform-rotate-x-general': 0,
		'transform-rotate-y-general': 0,
		'transform-rotate-z-general': 10,
		'transform-origin-x-general': 'right',
		'transform-origin-y-general': 'top',
		'font-family-general': 'Montserrat',
		'font-size-unit-general': 'px',
		'letter-spacing-unit-general': 'px',
		'line-height-unit-general': 'px',
		'text-shadow-general':
			'2px 4px 3px rgba(var(--maxi-light-color-8),0.3)',
		'margin-sync-horizontal-general': false,
		'margin-sync-vertical-general': false,
		'padding-sync-horizontal-general': false,
		'padding-sync-vertical-general': false,
		'palette-color-general': 3,
		'palette-color-general-hover': 5,
		'palette-status-general': true,
		'palette-status-general-hover': true,
		'transform-origin-x-unit-general': '%',
		'transform-origin-y-unit-general': '%',
		'transition-duration-general': 0.3,
		'font-size-xl': 20,
		'custom-formats': {
			'maxi-text-block__custom-format--0': {
				'font-weight-general': 800,
			},
			'maxi-text-block__custom-format--1': {
				'font-style-general': 'italic',
			},
			'maxi-text-block__custom-format--2': {
				'color-general': 'rgba(49,16,215,1)',
			},
		},
		'typography-status-hover': false,
	},
	innerBlocks: [],
	validationIssues: [],
	originalContent:
		'<div id="text-maxi-5se8ef1z-u81582" class="maxi-block maxi-text-block maxi-motion-effect maxi-motion-effect-text-maxi-5se8ef1z-u81582 maxi-light text-maxi-5se8ef1z-u81582 wp-block-maxi-blocks-text-maxi maxi-sc-light-background-color-1 maxi-sc-light-border-color-4 maxi-sc-light-box-shadow-color-3" data-align="full"><div class="maxi-background-displayer"><div class="maxi-background-displayer__layer maxi-background-displayer__color"></div></div><p class="maxi-text-block__content" data-gx_initial_block_class="maxi-def-light"><span class="maxi-text-block--has-custom-format maxi-text-block__custom-format--2">Testing</span> <span class="maxi-text-block--has-custom-format maxi-text-block__custom-format--1">Copy</span> <span class="maxi-text-block--has-custom-format maxi-text-block__custom-format--0">Paste</span></p></div>',
};
