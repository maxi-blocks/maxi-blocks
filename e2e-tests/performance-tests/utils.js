/**
 * WordPress dependencies
 */
import { createNewPost } from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import { execSync } from 'child_process';
import {
	BLOCKS_LOAD_TIMEOUT,
	ITERATIONS,
	RESULTS_FILE_DIR,
	WARMUP_ITERATIONS,
} from './config';

/**
 * External dependencies
 */
import fs from 'fs';
import path from 'path';
import { Client } from 'typesense';

const DEBUG = process.env.DEBUG_PERF_TESTS === 'true';

export function debugLog(...args) {
	if (DEBUG) {
		console.debug(...args);
	}
}

/**
 * Wait for the blocks to load.
 *
 * @param {Page} page
 * @param {number} expectedBlocksCount
 * @param {number} maxWaitTime
 */
export async function waitForBlocksLoad(page, expectedBlocksCount) {
	debugLog(`Waiting for ${expectedBlocksCount} blocks to load...`);
	try {
		await page.waitForFunction(
			expectedCount => {
				const editor = document.querySelector(
					'.block-editor-block-list__layout'
				);
				if (!editor) {
					console.log('Editor not found');
					return false;
				}
				const blocks = editor.querySelectorAll('.maxi-block');
				const loader = editor.querySelector(
					'.maxi-blocks-content-loader__bar'
				);

				return (
					(!loader || loader.offsetParent === null) &&
					blocks.length >= expectedCount
				);
			},
			{ timeout: BLOCKS_LOAD_TIMEOUT },
			expectedBlocksCount
		);
		debugLog('Blocks loaded successfully');
	} catch (error) {
		const blocks = await page.evaluate(() => {
			const editor = document.querySelector(
				'.block-editor-block-list__layout'
			);
			if (!editor) return 'Editor not found';
			const blocks = editor.querySelectorAll('.maxi-block');
			return blocks.length;
		});
		console.error(
			`Timeout waiting for ${expectedBlocksCount} blocks to load, actual count: ${blocks}`
		);
		console.error(error);
		throw error;
	}
}

/**
 * Measure the time it takes to perform a single action.
 *
 * @param {Function} action
 * @param {any} context
 * @returns {Object<string, any>} { time: number, context: Any }
 */
export async function measureSingleAction(action, context) {
	const start = performance.now();
	const newContext = await action(context);
	const end = performance.now();
	return { time: end - start, context: newContext };
}

/**
 * Perform measurements for a set of events.
 *
 * @param {Object<string, {pre: Function, action: Function, post: Function}>} events
 * @param {number} iterations
 * @returns {Object<string, {times: number[], average: number}>} measurements
 */
export async function performMeasurements(events, iterations = ITERATIONS) {
	const results = Object.fromEntries(
		Object.keys(events).map(key => [key, { times: [] }])
	);

	for (let i = 0; i < iterations; i++) {
		debugLog(`Starting iteration ${i + 1} of ${iterations}`);
		await createNewPost();
		let context = {};
		for (const [key, event] of Object.entries(events)) {
			try {
				if (event.pre) {
					debugLog(`Pre-action for ${key}`);
					context = await event.pre(context);
				}
				debugLog(`Measuring action for ${key}`);
				const { time, context: newContext } = await measureSingleAction(
					event.action,
					context
				);
				results[key].times.push(time);
				context = newContext;
				if (event.post) {
					debugLog(`Post-action for ${key}`);
					context = await event.post(context);
				}
			} catch (error) {
				console.error(`Error in test for event ${key}:`, error);
				throw error;
			}
		}
		debugLog(`Iteration ${i + 1} completed`);
	}

	for (const key in results) {
		results[key].average =
			results[key].times.reduce((a, b) => a + b, 0) / iterations;
	}

	return results;
}

/**
 * Save the measurements for a single event.
 *
 * @param {string} key
 * @param {Object<string, {times: number[], average: number}>} measurements
 */
export function saveEventMeasurements(key, measurements) {
	debugLog(`Saving measurements for ${key}`);
	const resultsFileName = getResultsFileName();
	const resultsFilePath = path.join(RESULTS_FILE_DIR, resultsFileName);

	let results = {};
	try {
		fs.mkdirSync(RESULTS_FILE_DIR, { recursive: true });
		if (fs.existsSync(resultsFilePath)) {
			results = JSON.parse(fs.readFileSync(resultsFilePath, 'utf8'));
		}
		results[key] = measurements;
		fs.writeFileSync(resultsFilePath, JSON.stringify(results, null, 2));
	} catch (error) {
		console.error('Error saving measurements:', error);
	}
}

/**
 * Generate the results file name with the current branch
 */
export function getResultsFileName() {
	const hash = getCurrentHash();
	return `performance-results-${hash}.json`;
}

/**
 * Get the current git hash
 */
export function getCurrentHash() {
	try {
		return execSync('git rev-parse --short HEAD').toString().trim();
	} catch (error) {
		console.error('Error getting git hash:', error);
		return 'unknown';
	}
}

/**
 * Warmup runs to stabilize the environment.
 */
export async function warmupRun() {
	console.info(`Starting warmup (${WARMUP_ITERATIONS} iterations)`);
	for (let i = 0; i < WARMUP_ITERATIONS; i++) {
		debugLog(`Warmup iteration ${i + 1}`);
		await createNewPost();
	}
	console.info('Warmup completed');
}

/**
 * Returns callback, click on which will add block to the page
 *
 * @param {Page} page
 * @param {string} blockName
 * @returns {Promise<Function>} Callback to click on to insert block
 */
export async function prepareInsertMaxiBlock(page, blockName) {
	const block = await page.waitForSelector(
		'.block-editor-default-block-appender__content',
		{ visible: true }
	);
	await page.evaluate(block => {
		block.focus();
	}, block);
	await page.keyboard.type(`/${blockName}`);
	return async () => {
		const block = await page.waitForSelector(
			'.components-autocomplete__result',
			{ visible: true }
		);
		await page.evaluate(block => {
			block.click();
		}, block);
	};
}

export class PatternManager {
	constructor(page) {
		this.page = page;
		this.searchClient = this.createSearchClient();
	}

	async searchPatternByName(patternName) {
		debugLog(`Searching for pattern: ${patternName}`);
		const searchParameters = {
			q: patternName,
			query_by: 'post_title',
			filter_by: 'gutenberg_type:=[Patterns,Pages]',
			sort_by: '_text_match:desc',
			per_page: 1,
		};

		try {
			const searchResults = await this.searchClient
				.collections('post')
				.documents()
				.search(searchParameters);

			if (searchResults.hits.length > 0) {
				return searchResults.hits[0].document;
			}
			return null;
		} catch (error) {
			console.error('Error searching for pattern:', error);
			return null;
		}
	}

	async getPatternCodeEditor(patternName) {
		const pattern = await this.searchPatternByName(patternName);
		if (pattern) {
			debugLog(`Pattern found: ${pattern.post_title}`);
			return pattern.gutenberg_code;
		} else {
			console.warn(`Pattern not found: ${patternName}`);
		}
	}

	createSearchClient() {
		return new Client({
			nodes: [
				{
					host: process.env.REACT_APP_TYPESENSE_API_URL,
					port: '443',
					protocol: 'https',
				},
			],
			apiKey: process.env.REACT_APP_TYPESENSE_API_KEY,
		});
	}
}
