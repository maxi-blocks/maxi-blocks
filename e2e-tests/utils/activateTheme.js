/**
 * WordPress dependencies
 */
import { activateTheme as wpActivateTheme } from '@wordpress/e2e-test-utils';

/**
 * Activates a theme with retry logic for CI environments.
 *
 * @param {string} slug Theme slug.
 */
export default async function activateTheme(slug) {
	const maxRetries = 3;
	let attempt = 1;

	// eslint-disable-next-line no-constant-condition
	while (true) {
		try {
			// eslint-disable-next-line no-await-in-loop
			await wpActivateTheme(slug);
			break; // Success, exit retry loop
		} catch (error) {
			if (attempt === maxRetries) {
				throw error; // Final attempt failed, throw error
			}
			// eslint-disable-next-line no-await-in-loop
			await new Promise(resolve => {
				setTimeout(resolve, 2000);
			}); // Wait 2s before retry
			attempt += 1;
		}
	}
}
