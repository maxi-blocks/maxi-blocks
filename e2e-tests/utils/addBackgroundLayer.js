import openSidebarTab from './openSidebarTab';

const addBackgroundLayer = async (page, type, isHover = false) => {
	const accordion = await openSidebarTab(page, 'style', 'background layer');
	await accordion.$$eval(
		'.maxi-tabs-control button',
		(button, _isHover) => button[+_isHover].click(),
		isHover
	);

	const selector = await page.$('.maxi-background-control__add-layer select');

	await selector.select(type);
};

export default addBackgroundLayer;
