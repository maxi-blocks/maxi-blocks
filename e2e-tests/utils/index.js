export { checkSCResult, receiveSelectedMaxiStyleCard } from './checkSCResult';
export { default as addBackgroundLayer } from './addBackgroundLayer';
export { default as addBackgroundLayerCanvas } from './addBackgroundLayerCanvas';
export { default as addCustomCSS } from './addCustomCSS';
export { default as addImageToImageMaxi } from './addImageToImageMaxi';
export { default as addImageToLibrary } from './addImageToLibrary';
export { default as addResponsiveTest } from './addResponsiveTest';
export { default as addTypographyOptions } from './addTypographyOptions';
export { default as addTypographyStyle } from './addTypographyStyle';
export { default as changeResponsive } from './changeResponsive';
export { default as copySCToEdit } from './copySCToEdit';
export { default as deactivatePlugin } from './deactivatePlugin';
export { default as editAdvancedNumberControl } from './editAdvancedNumberControl';
export { default as editAxisControl } from './editAxisControl';
export { default as editColorControl } from './editColorControl';
export { default as editGlobalStyles } from './editGlobalStyles';
export { default as getAttributes } from './getAttributes';
export { default as getBlockAttributes } from './getBlockAttributes';
export { default as getBlockStyle } from './getBlockStyle';
export { default as getClientId } from './getClientId';
export { default as getEditedPostContent } from './getEditedPostContent';
export { default as getElementAttribute } from './getElementAttribute';
export { default as getStyleCardEditor } from './getStyleCardEditor';
export { default as insertMaxiBlock } from './insertMaxiBlock';
export { default as modalMock } from './modalMock';
export { default as openPreviewPage } from './openPreviewPage';
export { default as openSidebarTab } from './openSidebarTab';
export { default as removeBackgroundLayers } from './removeBackgroundLayers';
export { default as resettingAttributes } from './resettingAttributes';
export { default as setAttributes } from './setAttributes';
export { default as waitForAttribute } from './waitForAttribute';
export { default as uniqueIDGenerator } from './uniqueIDGenerator';
export { default as updateAllBlockUniqueIds } from './updateAllBlockUniqueIds';
export { default as changeUniqueID } from './changeUniqueID';
