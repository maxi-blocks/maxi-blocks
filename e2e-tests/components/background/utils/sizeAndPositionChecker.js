import { editAdvancedNumberControl, editAxisControl } from '../../../utils';

const valuesDictionary = {
	normal: {
		base: {
			size: {
				number: '50',
				unit: 'px',
			},
			position: {
				values: ['1', '2', '3', '5'],
				unit: '%',
			},
		},
		m: {
			size: { number: '35', unit: 'vw' },
			position: { values: ['55', '89', '144', '233'], unit: 'em' },
		},
		s: {
			size: { number: '76', unit: 'px' },
			position: { values: ['67', '37', '12', '7'], unit: 'px' },
		},
		xs: {
			size: { number: '25', unit: 'em' },
			position: { values: ['8', '13', '21', '34'], unit: 'vw' },
		},
	},
	hover: {
		base: {
			size: { number: '75', unit: '%' },
			position: { values: ['2', '4', '7', '13'], unit: 'px' },
		},
		m: {
			size: { number: '23', unit: 'px' },
			position: {
				values: ['23', '29', '31', '37'],
				unit: 'px',
			},
		},
		s: {
			size: { number: '23', unit: 'px' },
			position: { values: ['24', '44', '81', '149'], unit: 'px' },
		},
		xs: {
			size: { number: '23', unit: 'px' },
			position: { values: ['11', '13', '17', '19'], unit: 'px' },
		},
	},
};

const sizeAndPositionChecker = async ({
	page,
	breakpoint = 'base',
	isHover = false,
}) => {
	const sizeClass = '.maxi-background-control__size';
	const positionClass = '.maxi-background-control__position';

	// size
	const size = await page.$(sizeClass);

	const {
		[isHover ? 'hover' : 'normal']: {
			[breakpoint]: {
				size: { number: sizeNumber, unit: sizeUnit },
				position: { values: positionValues, unit: positionUnit },
			},
		},
	} = valuesDictionary;

	await editAdvancedNumberControl({
		page,
		instance: size,
		newNumber: sizeNumber,
		newValue: sizeUnit,
	});

	// position
	const position = await page.$(positionClass);

	await editAxisControl({
		page,
		instance: position,
		syncOption: 'none',
		values: positionValues,
		unit: positionUnit,
	});

	return true;
};

export default sizeAndPositionChecker;
