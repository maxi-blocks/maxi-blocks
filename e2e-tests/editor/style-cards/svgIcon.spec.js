/**
 * WordPress dependencies
 */
import { createNewPost, setBrowserViewport } from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import { getStyleCardEditor, checkSCResult, copySCToEdit } from '../../utils';

describe('SC svg icon', () => {
	it('Checking svg icon accordion', async () => {
		await createNewPost();
		await setBrowserViewport('large');

		await getStyleCardEditor({
			page,
			accordion: 'SVG',
		});
		await copySCToEdit(page, `copy - ${Date.now()}`);

		// Global Line Colour
		await page.$eval(
			'.maxi-blocks-sc__type--SVG .maxi-style-cards-control__toggle-line-color-global input',
			button => button.click()
		);

		await page.$$eval(
			'.maxi-color-control__palette .maxi-color-control__palette-container button',
			button => button[4].click()
		);

		// Global Fill Colour
		await page.$eval(
			'.maxi-blocks-sc__type--SVG .maxi-style-cards-control__toggle-fill-color-global input',
			button => button.click()
		);

		await page.$$eval(
			'.maxi-color-control__palette .maxi-color-control__palette-container button',
			button => button[4].click()
		);

		expect(await checkSCResult(page)).toMatchSnapshot();
	});
});
