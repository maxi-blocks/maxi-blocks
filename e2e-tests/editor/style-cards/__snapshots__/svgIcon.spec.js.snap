// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SC svg icon Checking svg icon accordion 1`] = `
{
  "icon": {
    "fill-color-global": true,
    "line-color": "",
    "line-color-global": true,
    "line-palette-color": 5,
    "line-palette-opacity": 1,
    "line-palette-status": true,
  },
}
`;
